src/components/Dialog/Dialog.tsx

```diff
@@ -117,8 +140,22 @@ export default class Dialog extends Component<DialogProps, DialogState> {
   };
 
   async componentDidMount() {
-    const sources = await this.getSourceIDAndPaths();
-    this.setState({ allSources: sources });
+    // If the API key is not valid do not attempt to load sources
+    if (!this.state.verified) {
+      this.setState({
+        errors: [invalidApiKeyError()],
+      });
+      return;
+    }
+    try {
+      const sources = await this.getSourceIDAndPaths();
+      if (sources.length === 0) {
+        throw noSourcesError();
+      }
+      this.setState({ allSources: sources });
+    } catch (error) {
+      this.setState({ errors: [error] });
```

Comment:
I'm curious what happens if the app goes through this catch block, more specifically what error is reflected in the UI. If the only error the try block can catch is from `getSourceIDAndPaths` then I think it will end up setting the error state to an API error. But I could be wrong there.

Reply:
That's a good q. `getSourceIDAndPaths()` has its own try/catch that will keep its error from reaching this block. So I think in practice we'd either fail silently (just in the console) or only throw a `noSourcesError()`. IE, I don't think it's actually possible for an `APIError` to get thrown here.